umap/static/umap/js/components/alerts/alert.js

```diff
@@ -0,0 +1,177 @@
+      if (event.target.closest('[data-close]')) {
+        this._hide()
+      }
+    })
+    document.addEventListener('umap:alert', (event) => {
+      this._displayAlert(event.detail)
+    })
+  }
+}
+
+class uMapAlertCreation extends uMapAlert {
+  static info(
+    message,
+    // biome-ignore lint/style/useNumberNamespace: Number.Infinity returns undefined by default
+    duration = Infinity,
+    editLink = undefined,
+    sendLink = undefined
+  ) {
+    const event = new CustomEvent('umap:alert-creation', {
+      bubbles: true,
+      cancelable: true,
+      detail: { message, duration, editLink, sendLink },
+    })
+    document.dispatchEvent(event)
+  }
+
+  constructor() {
+    super()
+    this.linkWrapper = this.container.querySelector('#link-wrapper')
+    this.formWrapper = this.container.querySelector('#form-wrapper')
+  }
+
+  _displayCreationAlert(detail) {
+    const {
+      level = 'info',
+      duration = 5000,
+      message = '',
+      editLink = undefined,
+      sendLink = undefined,
+    } = detail
+    uMapAlert.prototype._displayAlert.call(this, { level, duration, message })
+    this.linkWrapper.querySelector('input[type="url"]').value = editLink
+    const button = this.linkWrapper.querySelector('input[type="button"]')
+    button.addEventListener('click', (event) => {
+      event.preventDefault()
+      L.Util.copyToClipboard(editLink)
+      event.target.value = L._('✅ Copied!')
+    })
+    if (sendLink) {
+      this.formWrapper.removeAttribute('hidden')
+      const form = this.formWrapper.querySelector('form')
+      form.addEventListener('submit', async (event) => {
+        event.preventDefault()
+        const formData = new FormData(form)
+        const server = new U.ServerRequest()
+        this._hide()
+        await server.post(sendLink, {}, formData)
+      })
+    }
+  }
+
+  connectedCallback() {
+    this.addEventListener('click', (event) => {
+      if (event.target.closest('[data-close]')) {
+        this._hide()
+      }
+    })
+    document.addEventListener('umap:alert-creation', (event) => {
+      this._displayCreationAlert(event.detail)
+    })
+  }
+}
+
+class uMapAlertChoice extends uMapAlert {
+  static error(
+    message,
+    // biome-ignore lint/style/useNumberNamespace: Number.Infinity returns undefined by default
+    duration = Infinity
+  ) {
+    const event = new CustomEvent('umap:alert-choice', {
+      bubbles: true,
+      cancelable: true,
+      detail: { level: 'error', message, duration },
+    })
+    document.dispatchEvent(event)
+  }
+
+  constructor() {
+    super()
+    this.choiceWrapper = this.container.querySelector('#choice-wrapper')
+  }
+
+  _displayChoiceAlert(detail) {
+    const { level = 'info', duration = 5000, message = '' } = detail
+    uMapAlert.prototype._displayAlert.call(this, { level, duration, message })
+    const form = this.choiceWrapper.querySelector('form')
+    form.addEventListener('submit', (event) => {
```

Comment:
Can't we just have have two buttons with their listeners instead of a form we loop on the `submitter.id` ? That would look more readable to me.